go/test/endtoend/vreplication/vstream_test.go

```diff
@@ -217,6 +214,10 @@ const vschemaSharded = `
 `
 
 func insertRow(keyspace, table string, id int) {
+	vtgateConn := getConnectionNoError(vc.t, vc.ClusterConfig.hostname, vc.ClusterConfig.vtgateMySQLPort)
+	if vtgateConn == nil {
+		return
+	}
```

Comment:
This doesn't feel right to me. I think that we should either return the error and handle it, or we should use the test reference to fail the test within the function using e.g. `require.NoError(t, err)`.

Reply:
Due to race conditions this call is sometimes made after vtgates have shutdown. This is not an error.